core/src/main/java/com/amplifyframework/storage/InvalidStorageBucketException.kt

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amplifyframework.storage
+
+import com.amplifyframework.AmplifyException
+import com.amplifyframework.annotations.InternalAmplifyApi
+
+/**
+ * Exception thrown when an invalid StorageBucket is specified.
+ */
+class InvalidStorageBucketException @InternalAmplifyApi constructor(
```

Comment:
the constructoer here should be marked `internal`, not that annotation.

I think this is a legitimate public error, but if not, the annotation would go on the class (not constructor) and that is only if you couldn't mark the class itself as internal, for various reasons.

Reply:
updated per suggestion.